src/java.base/share/classes/com/sun/crypto/provider/CipherCore.java

```diff
@@ -748,12 +748,26 @@ byte[] doFinal(byte[] input, int inputOffset, int inputLen)
         throws IllegalBlockSizeException, BadPaddingException {
         try {
             byte[] output = new byte[getOutputSizeByOperation(inputLen, true)];
-            byte[] finalBuf = prepareInputBuffer(input, inputOffset,
-                    inputLen, output, 0);
+            int outputOffset = 0;
+            int outLen = 0;
+
+            if (!decrypting && (padding != null) && (inputLen > 0)) {
+                /*
+                * Call the update() method to get rid of as many bytes as
+                * possible before potentially copying array to pad.
+                */
+                int updateOutLen = update(input, inputOffset, inputLen, output, outputOffset);
+                inputOffset = inputLen;
+                inputLen = 0;
+                outputOffset += updateOutLen;
+                outLen = updateOutLen;
```

Comment:
`outputOffset` is known to be zero here, and the extra local `updateOutLen` is unnecessary
```java
                outLen = update(input, inputOffset, inputLen, output, 0);
                outputOffset += outLen;
                inputOffset = inputLen;
                inputLen = 0;
```

Reply:
You are right, but I wrote it that way to make the code more readable. It provides a clearer view into what happens in my opinion.

I could change it, but I feel it's easier to understand with the extra variable. Do you think that's OK, or should I go with your suggestion?